java/code/src/com/suse/manager/utils/SaltUtils.java

```diff
@@ -1097,14 +1097,10 @@ private void handleImagePackageProfileUpdate(ImageInfo imageInfo,
                 Long instantNow = new Date().getTime() / 1000L;
                 OSImageInspectSlsResult ret = result.getKiwiInspect().getChanges().getRet();
                 List<OSImageInspectSlsResult.Package> packages = ret.getPackages();
-                packages.stream().forEach(pkg -> {
-                    createImagePackageFromSalt(pkg.getName(),
-                            pkg.getEpoch().equals(StringUtils.EMPTY) ? null :
-                                    pkg.getEpoch(),
-                            pkg.getRelease(), pkg.getVersion(),
-                            Optional.of(instantNow),
-                            Optional.of(pkg.getArch()), imageInfo);
-                });
+                packages.forEach(pkg -> createImagePackageFromSalt(pkg.getName(),
+                        PackageEvrFactory.lookupOrCreatePackageEvr(StringUtils.trimToNull(pkg.getEpoch()),
```

Comment:
Nitpick: Is there a particular reason to move `lookupOrCreatePackageEvr` here, instead of leaving it in `createImagePackageFromSalt`? I somewhat like the idea `createImagePackageFromSalt` is the method responsible for writing into the database.

Reply:
The aim was to keep the string sanitation in a single place.

I restructured the functions. Just a little more verbose, but should be more readable now.